esi_ui/api/esi_api.py

```diff
@@ -2,81 +2,122 @@
-    token = request.user.token.id
+    connection = esiclient(token=request.user.token.id)
+
+    def node_list_full_info(connection):
+        """Get a list of nodes with information from Ironic and ESI-Leap
+
+        :param connection: An ESI connection
+        :type connection: :class:`~esi.connection.ESIConnection`
+        :returns: A list of dictionaries containing baremetal and lease information
+        """
+
+        esi_nodes = []
+        baremetal_nodes = []
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            f1 = executor.submit(connection.lease.nodes)
+            f2 = executor.submit(connection.baremetal.nodes, details=True)
+            esi_nodes = list(f1.result())
+            baremetal_nodes = list(f2.result())
+
+        esi_nodes.sort(key=lambda node: node.name)
+        baremetal_nodes.sort(key=lambda node: node.name)
+
+        return [
+            {
+                'uuid': bm.id,
+                'name': bm.name,
+                'maintenance': bm.is_maintenance,
+                'provision_state': bm.provision_state,
+                'target_provision_state': bm.target_provision_state,
+                'power_state': bm.power_state,
+                'target_power_state': bm.target_power_state,
+                'properties': bm.properties,
+                'lease_uuid': e.lease_uuid,
+                'owner': e.owner,
+                'lessee': e.lessee,
+                'resource_class': e.resource_class,
+                'future_offers': e.future_offers,
+                'future_leases': e.future_leases,
+            }
+            for e, bm in zip(esi_nodes, baremetal_nodes)
+        ]
 
-    esi_nodes = []
-    ironic_nodes = []
     with concurrent.futures.ThreadPoolExecutor() as executor:
-        esi_future = executor.submit(esiclient(token).list_nodes)
-        ironic_future = executor.submit(ironicclient(token).node.list, detail=True)
-        esi_nodes = esi_future.result()
-        ironic_nodes = ironic_future.result()
-
-    esi_nodes.sort(key=lambda node: node.name)
-    ironic_nodes.sort(key=lambda node: node.name)
-
-    return [
-        {
-            "name": ironic_node.name,
-            "uuid": ironic_node.uuid,
-            "provision_state": ironic_node.provision_state,
-            "target_provision_state": ironic_node.target_provision_state,
-            "power_state": ironic_node.power_state,
-            "target_power_state": ironic_node.target_power_state,
-            "maintenance": ironic_node.maintenance,
-            "lease_uuid": esi_node.lease_uuid,
-            "owner": esi_node.owner,
-            "lessee": esi_node.lessee,
-            "resource_class": esi_node.resource_class,
-        }
-        for esi_node, ironic_node in zip(esi_nodes, ironic_nodes)
-    ]
-
-
-def set_power_state(request, node_id, state):
+        f1 = executor.submit(nodes.network_list, connection)
+        f2 = executor.submit(node_list_full_info, connection)
```

Comment:
My first instinct was to say that making `node_list_full_info` a separate function just increases the complexity of the code. But the longer-term plan is to push that retrieval into `esi node list`, at which point this will just be a single call, so that's fine. I think it's worth putting a comment in `node_list_full_info` that mentions that this is a temporary solution that will be updated once `esi-leap` is updated.